### ISSUE TITLE
[AIR 50.2.4.4+] `NetStream` fails to play local files which name starting from number

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#3087]
## Problem Description
`NetStream` fails to play local files which name starting from number using AIR 50.2.4.4+.

Tested with multiple AIR 50.2.4.4 and latest 51.0.0.2 versions, with different AIR applications, different architectures, devices, platforms and videos.
Same issue in all cases.
There is no such issue using AIR 50.2.4.1.
Didn't test with AIR 50.2.4.2 and AIR 50.2.4.3.
Files which filename starts with non-digit could be played without issues.

Related issues:
https://github.com/airsdk/Adobe-Runtime-Support/issues/1939
https://github.com/airsdk/Adobe-Runtime-Support/issues/82

## Steps to Reproduce
Launch application with code below with device. It just play video file `123.mp4`.

Application example with sources and video attached.
[video_filename_number_bug.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/14397572/video_filename_number_bug.zip)

```
package {
	import flash.display.Sprite;
	import flash.events.Event;
	import flash.net.NetConnection;
	import flash.net.NetStream;
	import flash.media.Video;
	import flash.events.NetStatusEvent;
	import flash.desktop.NativeApplication;
	import flash.desktop.SystemIdleMode;
	
	public class VideoFilenameNumberBug extends Sprite {
		private var nc:NetConnection;
		private var ns:NetStream;
		private var video:Video = new Video(640, 480);
		
		public function VideoFilenameNumberBug() {
			addEventListener(Event.ADDED_TO_STAGE, init);
		}
		
		private function init(e:Event):void {
			removeEventListener(Event.ADDED_TO_STAGE, init);
			
			NativeApplication.nativeApplication.systemIdleMode = SystemIdleMode.KEEP_AWAKE;
			addChild(video);
			
			nc = new NetConnection();
			nc.addEventListener(NetStatusEvent.NET_STATUS, ncHandler);
			nc.connect(null);
		}
		

		private function ncHandler(e:NetStatusEvent):void {
			if (e.info.code == "NetConnection.Connect.Success"){
				ns = new NetStream(nc);
				ns.client = {onMetaData:getMeta};
				ns.addEventListener(NetStatusEvent.NET_STATUS, nsHandler);
				video.attachNetStream(ns);
				ns.play("123.mp4");//This line cause SecurityError: Error #2147: Forbidden protocol in URL 123.mp4.
			}
		}

		private function nsHandler(e:NetStatusEvent):void {
			trace(e.info.code);
			if (e.info.code == "NetStream.Play.Stop"){
				ns.play("123.mp4");
			}
		}

		private function getMeta(mdata:Object):void { }
		
	}
}
```

**Actual Result:**
`NetStream` failed with error:
```
*** Security Sandbox Violation ***
Connection to 123.mp4 halted - not permitted from app:/video_filename_number_bug.swf
SecurityError: Error #2147: Forbidden protocol in URL 123.mp4.
	at flash.net::NetStream/play()
```

**Expected Result:**
Video playback without errors.

## Known Workarounds
1) Use AIR 50.2.4.1 or earlier.
2) Use `File.applicationDirectory.resolvePath("123.mp4").nativePath` or `File.applicationDirectory.resolvePath("123.mp4").url`.
3) Do not use file with names starting with numbers.

**Comment by ajwfrost** [comment]
Looks like an error introduced when checking for an invalid URL scheme has also hit the scenario when there's no scheme .. should be able to get a fix fairly quickly.
thanks

**Comment by itlancer** [comment]
Fixed with latest AIR 51.0.0.3.
Thanks!
